data_steward/analytics/cdr_ops/report_runner.py

```diff
@@ -62,47 +148,50 @@ def display_notebook_help(notebook_path, params):
 #   new_rdr: Unknown type (default "")
 
 
-def main():
-    parser = argparse.ArgumentParser()
-    parser.add_argument('notebook_path', help='A .py jupytext file.')
-    parser.add_argument('output_path', default="", help='An output .html file')
-    parser.add_argument('--help_notebook', action='store_true')
-    parser.add_argument('--params', '-p', nargs=2, action='append')
-
-    args = parser.parse_args()
-    notebook_path = args.notebook_path
-    output_path = args.output_path
-    help_notebook = args.help_notebook
-    params = dict(args.params)
-
-    #Convert py to ipynb
-    create_ipynb_from_py(notebook_path)
-    notebook_path = PurePath(notebook_path).with_suffix('.ipynb')
+def main(notebook_jupytext_path, params, output_path, help_notebook=False):
+    """
+    
+    :param notebook_jupytext_path: 
+    :param params: 
+    :param output_path: 
+    :param help_notebook: 
+    :return: 
+    """
+    # Convert py to ipynb
+    surrogate_output_path = create_ipynb_from_py(notebook_jupytext_path)
 
     if help_notebook:
-        sys.exit(display_notebook_help(str(notebook_path), params))
+        sys.exit(display_notebook_help(surrogate_output_path))
 
-    surrogate_output_path = PurePath(output_path).with_suffix('.ipynb')
+    if not validate_notebook_params(surrogate_output_path, params):
+        display_notebook_help(surrogate_output_path)
+        sys.exit('Missing required parameters')
 
     try:
-        #Pass ipynb to papermill
-        execute_notebook(str(notebook_path),
-                         str(surrogate_output_path),
+        # Pass ipynb to papermill
+        execute_notebook(surrogate_output_path,
```

Comment:
Passing `surrogate_output_path` as the input and output of the execution will overwrite the ipynb version of the jupytext notebook, correct? Is that intended?